fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/FedoraToOCFLObjectIndexImpl.java

```diff
@@ -59,15 +86,39 @@ public FedoraOCFLMapping addMapping(final String fedoraResourceIdentifier, final
         if (mapping == null) {
             mapping = new FedoraOCFLMapping(fedoraRootObjectResourceId, ocflObjectId);
             fedoraOCFLMappingMap.put(fedoraRootObjectResourceId, mapping);
-
+            writeMappingToDisk(fedoraRootObjectResourceId, mapping);
         }
 
         if (!fedoraResourceIdentifier.equals(fedoraRootObjectResourceId)) {
             fedoraOCFLMappingMap.put(fedoraResourceIdentifier, mapping);
+            writeMappingToDisk(fedoraResourceIdentifier, mapping);
         }
 
         LOGGER.debug("added mapping {} for {}", mapping, fedoraResourceIdentifier);
         return mapping;
     }
 
+    /**
+     * Write any added mappings to the on-disk index.
+     * @param fedoraId The internal Fedora identifier.
+     * @param fedoraOCFLMapping The Fedora to OCFL mapping object.
+     */
+    private void writeMappingToDisk(final String fedoraId, final FedoraOCFLMapping fedoraOCFLMapping) {
+        try {
+            if (!FEDORA_TO_OCFL_INDEX_FILE.exists()) {
+                final String dirName = FEDORA_TO_OCFL_INDEX_FILE.toPath().toAbsolutePath().toString().substring(0,
```

Comment:
Instead of these contortions, can you try either:
- https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html#getParent() or
- https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html#getParentFile()